src/HealthChecks/HealthChecks/src/HealthCheckPublisherHostedService.cs

```diff
@@ -150,8 +170,34 @@ private async Task RunAsyncCore(CancellationToken cancellationToken)
         // Forcibly yield - we want to unblock the timer thread.
         await Task.Yield();
 
+        _timerElapsed += _healthCheckPublisherOptions.Value.Period;
+
+        // Filter health check registrations per elapsed
+        var healthCheckRegistrationsElapsed = new HashSet<string>();
+        foreach (var registration in _healthCheckRegistrationsDictionary.Values)
+        {
+            if (_timerElapsed.Ticks % registration.Period.Ticks == 0L)
```

Comment:
We may want to test if `Timespan.Ticks` would give us enough precision for testing the periodicity of the HC.

Reply:
Another alternative would be to keep the `lastTriggeredTime` TimeSpan in memory for each HC, and then test if `now - lastTriggeredTime[HC] > HC_Period`.